app/src/lib/release-notes.ts

```diff
@@ -0,0 +1,105 @@
+const ordinal: (i: number) => string = require('ordinal')
+
+import {
+  ReleaseMetadata,
+  ReleaseNote,
+  ReleaseSummary,
+} from '../models/release-notes'
+
+const itemEntryRe = /^\[(new|fixed|improved|removed|added|pretext)\]\s(.*)/i
+
+const monthNames = [
+  'January',
+  'February',
+  'March',
+  'April',
+  'May',
+  'June',
+  'July',
+  'August',
+  'September',
+  'October',
+  'November',
+  'December',
+]
+
+function formatDate(date: Date) {
+  const day = date.getDate()
+  const dayWithSuffix = ordinal(day)
+  const monthIndex = date.getMonth()
+  const year = date.getFullYear()
+
+  return `${monthNames[monthIndex]} ${dayWithSuffix} ${year}`
+}
+
+export function parseReleaseEntries(
+  notes: ReadonlyArray<string>
+): ReadonlyArray<ReleaseNote> {
+  const entries = new Array<ReleaseNote>()
+
+  for (const note of notes) {
+    const text = note.trim()
+    const match = itemEntryRe.exec(text)
+    if (match === null) {
+      continue
+    }
+    const kind = match[1].toLowerCase()
+    const message = match[2]
+
+    switch (kind) {
+      case 'new':
+      case 'fixed':
+      case 'improved':
+      case 'removed':
+      case 'added':
+      case 'pretext':
+        entries.push({ kind, message })
+    }
+  }
+
+  return entries
+}
+
+export function getReleaseSummary(
+  latestRelease: ReleaseMetadata
+): ReleaseSummary {
+  const entries = parseReleaseEntries(latestRelease.notes)
+
+  const enhancements = entries.filter(
+    e => e.kind === 'new' || e.kind === 'added' || e.kind === 'improved'
+  )
+  const bugfixes = entries.filter(e => e.kind === 'fixed')
+  const other = entries.filter(e => e.kind === 'removed')
+
+  const publishedDate = new Date(latestRelease.pub_date)
+
+  return {
+    latestVersion: latestRelease.version,
+    datePublished: formatDate(publishedDate),
+    // TODO: find pretext entry
```

Comment:
what does this mean?

Reply:
https://github.com/desktop/desktop/issues/5510